### PR TITLE
Cap aim level when there is no target or when the target is not visible

### LINKED ISSUE #38199: Turrets are vulnerable to blind fire
# Describe the bug

It is very easy to defeat any turret if you crouch behind an object and blind-fire to where the turret is.

# Steps To Reproduce

1. Find a turret holding a hoard of ammo and protecting a valuable place
2. Find a table
3. Push table in front of you and approact turret
4. Crouch and "aim" "precisely" at a tile where the turret is
5. Depending on your weapon, turret will eventually be dead.

# Expected behavior

Turret shoots your table to shreds, or you can't aim at all, or it shoots your hands and weapon while you aim 

# Screenshots

# Versions and configuration
Latest master.

# Additional context

### PR DESCRIPTION
**Pull request by kevingranade**
#### Summary
SUMMARY: Balance "Nerf blindfire with ranged weapons."

#### Purpose of change
Fixes #38199 
Turrets already returned fire when fired upon, this greatly reduces your chance of hitting them when blindfiring.
More generally, blindfiring with good accuracy didn't make sense in general.

#### Describe the solution
Caps achievable accuracy to being capable of reliably hitting *something* in the target tile when blindfiring.

#### Describe alternatives you've considered
Disallowing aiming entirely, yielding extremely poor accuracy when blindfiring.
Capping accuracy to a fixed angle, allowing relatively accurate fire when near a target, doesn't really make sense as the actual location of the target is unknown.
Reducing accuracy even further based on range so that distant shots don't guarantee the same effective accuracy as nearby shots.

#### Testing
Blindfolded a character and explored achievable aim level at various ranges.
Blind fired at a turret as outlined in #38199 and was filled with holes for my trouble.
2020-02-23-034148_1366x768_scrot

### CODE REVIEW

**Comment by Dacendeth** [comment]
Will this have any effect on aiming at targets you cant see in the dark, but can see with thermals on?

File: src/ranged.cpp [hunk 1]
```diff
@@ -1022,6 +1022,20 @@ static int print_ranged_chance( const player &p, const catacurses::window &w, in
     return line_number;
 }
 
+// Handle capping aim level when the player cannot see the target tile or there is nothing to aim at.
+static double calculate_aim_cap( const player &p, const tripoint &target )
+{
+    double min_recoil = 0.0;
+    if( !p.sees( target ) || g->critter_at( target, true ) == nullptr ) {
```
Comment (kevingranade) [inline src/ranged.cpp]:
```suggestion
    const Creature *victim = g->critter_at( target, true );
    if( victim == nullptr || !p.sees( victim ) {
```

**Comment by kevingranade** [comment]
I rerouted it through character::sees( creature ) , which will apply any special senses and such that are supported.

File: src/ranged.cpp [hunk 2]
```diff
@@ -1022,6 +1022,21 @@ static int print_ranged_chance( const player &p, const catacurses::window &w, in
     return line_number;
 }
 
+// Handle capping aim level when the player cannot see the target tile or there is nothing to aim at.
+static double calculate_aim_cap( const player &p, const tripoint &target )
+{
+    double min_recoil = 0.0;
+    const Creature *victim = g->critter_at( target, true );
+    if( victim == nullptr || !p.sees( victim ) {
```
Comment (kevingranade) [inline src/ranged.cpp]:
```suggestion
    if( victim == nullptr || !p.sees( *victim ) {
```

File: src/ranged.cpp [hunk 3]
```diff
@@ -1022,6 +1022,21 @@ static int print_ranged_chance( const player &p, const catacurses::window &w, in
     return line_number;
 }
 
+// Handle capping aim level when the player cannot see the target tile or there is nothing to aim at.
+static double calculate_aim_cap( const player &p, const tripoint &target )
+{
+    double min_recoil = 0.0;
+    const Creature *victim = g->critter_at( target, true );
+    if( victim == nullptr || !p.sees( *victim ) {
```
Comment (kevingranade) [inline src/ranged.cpp]:
```suggestion
    if( victim == nullptr || !p.sees( *victim ) ) {
```

**Comment by Voker57** [comment]
reliably hitting something in the target square while blind might still be too much, but at least you can't kite turrets now with slings, thanks!

**Pull request merged**